File: Backend/TailorApp.API/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TailorApp.API.Models;

public class Order
{
    public int OrderId { get; set; }
    public string OrderCode { get; set; } = string.Empty;
    public int CustomerId { get; set; }
    public int ShopId { get; set; }
    public int? MeasurementId { get; set; }

    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public DateTime? DeliveryDate { get; set; }

    public string Status { get; set; } = "Pending";

    [Column(TypeName = "decimal(10,2)")]
    public decimal TotalAmount { get; set; } = 0;

    [Column(TypeName = "decimal(10,2)")]
    public decimal Discount { get; set; } = 0;

    [Column(TypeName = "decimal(10,2)")]
    public decimal FinalAmount { get; set; } = 0;

    public string? Notes { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public Customer? Customer { get; set; }
    public Measurement? Measurement { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: Frontend/tailor-app-ui/src/app/models/order.ts
export interface OrderItem {
  orderItemId?: number;
  orderId?: number;
  garmentType: string;
  description?: string;
  quantity: number;
  unitPrice: number;
  totalPrice?: number;
  fabricId?: number;
  fabricName?: string;
  fabricDetails?: string;
  specialInstructions?: string;
}

export interface Order {
  orderId?: number;
  orderCode?: string;
  customerId: number;
  customerName?: string;
  customerPhone?: string;
  shopId: number;
  orderDate?: string;
  deliveryDate?: string;
  status?: string;
  totalAmount?: number;
  discount?: number;
  finalAmount?: number;
  measurementId?: number;
  notes?: string;
  createdDate?: string;
  orderItems: OrderItem[];
}

export const ORDER_STATUSES = [
  'Pending',
  'InProgress',
  'ReadyForDelivery',
  'Delivered',
  'Cancelled'
];